scmail.py

```diff
@@ -0,0 +1,205 @@
+    fingerprint = ctx.obj['gpg'].list_keys(True).curkey.get('fingerprint')
+    data = {'fingerprint': fingerprint}
+
+    # Register
+    address = SECUREMAILBOX_URL + '/register/'
+    r = requests.post(address)
+    print(r.text)
+
+    if 'success' in r:
+        print('Registration success.')
+    else:
+        print('Registration fail.\nError is: {}'.format(r.get("error")))
+
+
+@click.command()
+@click.option('--recipient', prompt='The recipient', required=True, help='The email address of the recipient.')
+@click.password_option('--password', prompt='Enter passwordo of private key to decrypt', help='The passphrase of private key')
+@click.pass_context
+def retrieve(ctx, recipient, password):
+    """Retrieve and Post messages from API"""
+    # Retrieve
+    address = SECUREMAILBOX_URL + '/retrieve/'
+    r = requests.post(address)
+    # Get response
+    res = json.loads(r)
+    # print(res)
+    if 'success' in res:
+        print('The message retrieve successful.')
+    else:
+        print('The message retrieve fail.\nError is: {}'.format(res.get("error")))
+        # return
+
+    # Decrypt the messages.
+    ok, messages = ctx.obj['gpg'].decrypt_message(messages=res, passphrase=password)
+    ctx.obj['pp'].pprint(messages[:-1])
+    if ok is False:
+        print('Other message decrypt fail.\n{}'.format(messages[-1]))
+    else:
+        ctx.obj['pp'].pprint(messages[-1])
+        print('Decrypt message successful.')
+
+
+@click.command()
+@click.option('--recipient', prompt='The recipient email', required=True, help='The email of the recipient.')
+@click.option('--message', required=True, prompt='The message', help='Message that will send.')
+@click.pass_context
+def send(ctx, recipient, message):
+    """Send a message.
+
+    recipient: the fingerprint of recipient
+
+    message: the messages.
+    """
+    # Choose recipient and Encrypted
+    ok, encrypted_message = ctx.obj['gpg'].encrypt_message(message, recipient)
+    if ok is False:
+        print('Message encrypt fail.\n{}'.format(encrypted_message))
+    else:
+        print('Message encrypt success.')
+
+    # Send
+    payload = {'fingerprint': recipient, 'message': message}
+    address = SECUREMAILBOX_URL + '/send/'
+    r = requests.post(address, json=payload)
+    r = json.loads(r)
+    print(r.text)
+
+
+@click.command()
+@click.option('--fingerprint', prompt='Enter the fingerprint of that key', required=True, help='The fingerprint of exported key.')
+@click.option('--is-file', prompt='Do you want to store keys to file (or print)?', is_flag=True, help='Whether export to file.')
+@click.option('--is-pvt', prompt='Do you want to export private key?', is_flag=True, help='Whether export private keys.')
+@click.pass_context
+def export_key(ctx, fingerprint, is_file, is_pvt):
+    file_name = None
+    passphrase = None
+
+    if is_file is True:
+        file_name = click.prompt('Enter the file name', type=str)
+        file_name = Path(file_name)
```

Comment:
What happens if the user inputs an invalid path and `Path` try's to parse it?
As well, `.touch()` doesn't complain if the file already exists so checking is superfluous. I.e. this could be shortened to:
```python
file_name = Path(file_name).touch()
```

Reply:
Thanks, I will change it to:

```python
file_name = Path(file_name).touch(exist_ok=False)
```

and add error checking.